internal/resources/providers/gcplib/inventory/provider.go

```diff
@@ -405,19 +400,21 @@ func getAssetsByProject[T any](assets []*ExtendedGcpAsset, log *logp.Logger, f T
 	return enrichedAssets
 }
 
-func getAllAssets(log *logp.Logger, it Iterator) []*assetpb.Asset {
+func (p *Provider) getAllAssets(ctx context.Context, request *assetpb.ListAssetsRequest) []*assetpb.Asset {
+	p.log.Infof("Listing asset types: %v of type %v for %v", request.AssetTypes, request.ContentType, request.Parent)
 	results := make([]*assetpb.Asset, 0)
+	it := p.inventory.ListAssets(ctx, request)
 	for {
 		response, err := it.Next()
 		if err == iterator.Done {
 			break
 		}
 		if err != nil {
-			log.Errorf("Error fetching GCP Asset: %s", err)
-			return nil
+			p.log.Errorf("Error fetching GCP Asset: %s", err)
+			return results
```

Comment:
That's great and I can see that eventually this is being appended into assets, we just need to make sure that there is no hidden logic that differentiates between `nil` and actually getting results.
Because we do not want to act is if everything succeeded when we have a partial response.
From what I saw in the code we act in a opportunistic sort of way, but I would consider in the future to recognize when we have partial results and potentially act on it instead of skipping the whole cycle (not in the scope of this PR).

Reply:
we still log the error like we did before, and the cycle continues without interruption like we did before, it just has a bit more findings to report. overall i think this change is safe, as returning `nil` or an empty `[]*assetpb.Asset` is the same in the sense that we never differentiate between the two, we just don't iterate on an empty value.